src/python/gem5/resources/resource.py

```diff
@@ -972,6 +938,234 @@ def obtain_resource(
+    :param resource_directory: The resource directory.
+    :param download_md5_mismatch: If the resource is present, but does not have
+                                  the correct md5 value, the resource will be
+                                  deleted and re-downloaded if this value is ``True``.
+                                  Otherwise an exception will be thrown.
+    :param clients: A list of clients to search for the resource. If this
+                    parameter is not set, it will default search all clients.
+    :param gem5_version: The gem5 version to use to filter incompatible
+                         resource versions. By default set to the current gem5
+                         version.
+    :param quiet: If ``True``, suppress output. ``False`` by default.
+    """
+    params = {}
+
+    db_query = []
+    for resource in workload["resources"].values():
+        db_query.append(
+            ClientQuery(
+                resource_id=resource["id"],
+                resource_version=resource["resource_version"],
+                gem5_version=gem5_version,
+            )
+        )
+    # Fetching resources as a list of dicts
+    resource_details_list = get_multiple_resource_json_obj(db_query, clients)
+
+    # Creating the resource objects for each resource
+    for param_name, param_resource in workload["resources"].items():
+        resource_match = None
+        for resource in resource_details_list:
+            if (
+                param_resource["id"] == resource["id"]
+                and param_resource["resource_version"]
+                == resource["resource_version"]
+            ):
+                resource_match = resource
+                break
+
+        if resource_match is None:
+            raise Exception(
+                f"Resource {param_resource['id']} with version {param_resource['resource_version']} not found"
+            )
+        assert isinstance(param_name, str)
+        to_path, downloader = _get_to_path_and_downloader_partial(
+            resource_json=resource_match,
+            to_path=local_path,
+            resource_directory=resource_directory,
+            download_md5_mismatch=download_md5_mismatch,
+            clients=clients,
+            gem5_version=gem5_version,
+            quiet=quiet,
+        )
+
+        resource_class = _get_resource_json_type_map[
+            resource_match["category"]
+        ]
+
+        params[param_name] = resource_class(
+            local_path=to_path,
+            downloader=downloader,
+            **resource,
+        )
+
+        # Adding the additional parameters to the workload parameters
+        if workload["additional_params"]:
+            for key in workload["additional_params"].keys():
+                assert isinstance(key, str)
+                value = workload["additional_params"][key]
+                params[key] = value
+
+    return WorkloadResource(
+        local_path=local_path,
+        downloader=None,
```

Comment:
Are we not passing the downloader?

Reply:
No, as the workload resource itself has no files to download, it just has resources that we need to download and those will be downloaded when we create those specific resource objects.

Reply:
Ack.